transformer_engine/jax/flax/transformer.py

```diff
@@ -637,6 +637,50 @@ def canonicalize_group_method(gm):
     return consecutive_impl()
 
 
+class LoRAScope:    # pylint: disable=too-few-public-methods
+    """LoRA Scope"""
+
+    def __init__(self, qkv_proj=False, output_proj=False, mlp=False):
+        self.qkv_proj = qkv_proj
+        self.output_proj = output_proj
+        self.mlp = mlp
+
+    def __eq__(self, other):
+        return (self.qkv_proj, self.output_proj, self.mlp) == \
+               (other.qkv_proj, other.output_proj, other.mlp)
+
+
+def _canonicalize_lora_scope(scope):
+    scope = scope.lower()
+
+    SCOPE_NONE = 'none'
+    SCOPE_ALL = 'all'
+    SCOPE_QKV_PROJ = 'qkv_proj'
+    SCOPE_OUTPUT_PROJ = 'output_proj'
+    SCOPE_MLP = 'mlp'
+    SCOPE_EX_QKV_PROJ = 'exclude_qkv_proj'
+    SCOPE_EX_OUTPUT_PROJ = 'exclude_output_proj'
+    SCOPE_EX_MLP = 'exclude_mlp'
+
+    assert scope in [
```

Comment:
I noticed that the `low_rank_adaptation_scope` is expected to be a string. Can we enhance the handle of `None` as other arguments accept `None` and I afrarid the users will be confused to use `None` or string `'None'`. I think we can either accepting `None` and convert it to string `'none'` or showing a error message to let user know they need to pass string `'none'` is okay.